s3box_review/s3_helpers.go

```diff
@@ -0,0 +1,102 @@
+package s3box
+
+import (
+	"bytes"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"sync"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/aws/aws-sdk-go/service/s3/s3manager"
+)
+
+const aesAlgo = "AES256" // Algo used for server-side encryption.
+
+// Modularize functions for testing
+var (
+	getRegionForBucket func(string) (string, error)
+	writeToS3          func(s3Handler *s3.S3, bucket string, fileKey string, data []byte, gzip bool) error
+)
+
+// getRegionForBucketProd looks up the region name for the given bucket
+func getRegionForBucketProd(name string) (string, error) {
+	// Any region will work for the region lookup, but the request MUST use PathStyle
+	config := aws.NewConfig().WithRegion("us-west-1").WithS3ForcePathStyle(true)
+	session := session.New()
```

Comment:
This is a great question. The logic is currently written so that only one s3 call should ever happen at once. However if we opened this up in the future, we'd probably want to clone the sessions. FWIW, I've not seen a closed AWS session in Clever code, however `Copy` is an available method.